collections-lists/FastestWayToCheckIfListIsOrdered/FastestWayToCheckIfListIsOrdered/ListOrderValidator.cs

```diff
@@ -0,0 +1,202 @@
+﻿using System.Runtime.InteropServices;
+
+namespace FastestWayToCheckIfListIsOrdered;
+
+public class ListOrderValidator
+{
+	public static bool IsOrderedUsingForLoop<T>(IList<T> list, IComparer<T>? comparer = default)
+	{
+		if (list.Count <= 1) return true;
+
+		comparer ??= Comparer<T>.Default;
+		var length = list.Count;
+		for (var i = 1; i < length; i++)
+		{
+			if (comparer.Compare(list[i - 1], list[i]) > 0) return false;
+		}
+
+		return true;
+	}
+
+	public static bool IsOrderedUsingArraySort<T>(IList<T> list, T[] array, IComparer<T>? comparer = default)
+	{
+		if (list.Count <= 1) return true;
+
+		comparer ??= Comparer<T>.Default;
+
+		var length = list.Count;
+
+		list.CopyTo(array, 0);
+		Array.Sort(array, 0, length, comparer);
+
+		for (var i = 0; i < length; i++)
+		{
+			if (comparer.Compare(list[i], array[i]) != 0) return false;
+		}
+
+		return true;
+	}
```

Comment:
The array creation should take place within this method. I see in the Program.cs that we are using a pooled array, which is exactly what we should be doing, but that is an implementation detail that should be hidden from the caller. The way the method is written now, we should be validating the length of the provided array, etc.
This would also have the added benefit of making the method signature the same as the other methods.